### PR TITLE
OMEdit: Add support for LibXml2 instead of XmlPatterns

### LINKED ISSUE #11558: QtXmlPatterns is deprecated
QtXmlPatterns module is deprecated and no longer available in qt 6.x
Luckily its usage is confined to OMEdit/OMEditLIB/Util/Utilities.cpp:

```
void Utilities::parseCompositeModelText(MessageHandler *pMessageHandler, QString contents)
{
  QFile schemaFile(QString(":/Resources/XMLSchema/tlmModelDescription.xsd"));
  schemaFile.open(QIODevice::ReadOnly);
  const QString schemaText(QString::fromUtf8(schemaFile.readAll()));
  schemaFile.close();
  const QByteArray schemaData = schemaText.toUtf8();
 
  QXmlSchema schema;
  schema.setMessageHandler(pMessageHandler);
  schema.load(schemaData);
  if (!schema.isValid()) {
    pMessageHandler->setFailed(true);
  } else {
    QXmlSchemaValidator validator(schema);
    if (!validator.validate(contents.toUtf8())) {
      pMessageHandler->setFailed(true);
    }
  }
}
```

it would need to be replaced with another xml library that supports DTD validation
could we use expat (already a dependency of FMILib) ? it would have to be compiled with DTD support enabled though

### PR DESCRIPTION
**Pull request by jschueller**
This adds an option OM_OMEDIT_ENABLE_LIBXML2 to use libxml2 instead of QtXmlPatterns for the DTD validation part
(we keep QXml for other xml stuff, this is not deprecated)

Closes #11558

The long-term goal is to support Qt6 (see also #11524 )

### CODE REVIEW

File: OMEdit/OMEditLIB/Util/Utilities.cpp [hunk 1]
```diff
@@ -606,6 +612,40 @@ void Utilities::parseCompositeModelText(MessageHandler *pMessageHandler, QString
...
+        xmlFreeDtd(dtd);
+      }
+      else
+      {
+        xmlFreeDtd(dtd);
+        pMessageHandler->setFailed(true);
+      }
+    }
+    else
+      pMessageHandler->setFailed(true);
```
Comment (perost) [inline OMEdit/OMEditLIB/Util/Utilities.cpp:643]: Missing `xmlFreeDoc(doc)` here?

**Comment by jschueller** [comment]
hi @adeas31, I hope you'll find time to get to this eventually

**Comment by jschueller** [comment]
hi @adeas31, I hope you'll find time to get to this eventually

**Comment by adeas31** [comment]
I will come back to this after 1.22.2 release.

**Comment by jschueller** [comment]
hi @adeas31, still busy ?

**Comment by adeas31** [comment]
Yeah still more important stuff going on. I have this on my todo list and we should have this in 1.23.0

**Review by adeas31 — APPROVED** [review]
LGTM.

As I understand, this will not give you error source like line number etc. If we can get it then it will be very good.

**Comment by adeas31** [comment]
@jschueller any update on this?

**Comment by jschueller** [comment]
No plans to add line number right now, can it still be merged though ?

**Pull request merged**